Add snap support for linux

Jan200101 opened the pull request:
Steam exists on snap! (unsure if in an official capacity, but its packaged by Canonical).

I put the code after the generic `$HOME/.steam/steam` check, because in the small testing I did Proton was completely unusable so any other released should be preferred.

Jan200101 commented:
Lets pretend I didn't have to force push to stop Clippy from beating me over the head twice over and once more just to get the format nice and tidy................

CosmicHorrorDev commented:
Thanks for the PR! I'm about to move, so I won't be getting around to reviewing till mid July at the earliest

Jan200101 commented:
No worries, this has no priority for anything.

Jan200101 commented:
Would it be better to rebase this onto v2 instead?

Jan200101 commented:
Bump.

Any update on this?
If needed I can give a step by step guide on how to test the functionality.

CosmicHorrorDev commented:
Forgot, thanks for the reminder! I'll write a note to take a look at this tomorrow. Would you mind switching the target to the dev v2 branch?

And I should be good for testing. I'll just spin up a VM to make sure everything works as expected

Jan200101 commented:
rebased and reworked for the dev v2 branch, verfied that the snapp installation ships with the same paths as flatpak and used those.

When testing be aware that the built-in Ubuntu Store had 2 Steam packages: "Steam Installer" and "Steam"

CosmicHorrorDev reviewed (approved):
Everything looks great, thanks!
